From 1e237fbebc414e23218350bdbb2b4f128f345800 Mon Sep 17 00:00:00 2001
From: Fam Zheng <famz@redhat.com>
Date: Wed, 9 Oct 2013 09:50:23 +0200
Subject: [PATCH 11/11] qemu-iotests: add monolithicFlat creation test to 059

RH-Author: Fam Zheng <famz@redhat.com>
Message-id: <1381312223-7074-4-git-send-email-famz@redhat.com>
Patchwork-id: 54795
O-Subject: [RHEL-7 qemu-kvm PATCH 3/3] qemu-iotests: add monolithicFlat creation test to 059
Bugzilla: 1017049
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

Signed-off-by: Fam Zheng <famz@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit 2fe2e2907163f6d86b6bbced776ec8f9319ca83f)
Signed-off-by: Fam Zheng <famz@redhat.com>
---
 tests/qemu-iotests/059     | 5 +++++
 tests/qemu-iotests/059.out | 7 +++++++
 2 files changed, 12 insertions(+)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 tests/qemu-iotests/059     |    5 +++++
 tests/qemu-iotests/059.out |    7 +++++++
 2 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/tests/qemu-iotests/059 b/tests/qemu-iotests/059
index b03429d..d2b3f9e 100755
--- a/tests/qemu-iotests/059
+++ b/tests/qemu-iotests/059
@@ -66,6 +66,11 @@ poke_file "$TEST_IMG" "$capacity_offset" "\xff\xff\xff\xff"
 poke_file "$TEST_IMG" "$grain_table_size_offset" "\x01\x00\x00\x00"
 { $QEMU_IO -c "read 0 512" $TEST_IMG; } 2>&1 | _filter_qemu_io | _filter_testdir
 
+echo "=== Testing monolithicFlat creation and opening ==="
+echo
+IMGOPTS="subformat=monolithicFlat" _make_test_img 2G
+$QEMU_IMG info $TEST_IMG | _filter_testdir
+
 # success, all done
 echo "*** done"
 rm -f $seq.full
diff --git a/tests/qemu-iotests/059.out b/tests/qemu-iotests/059.out
index 9e715e5..2a4c44d 100644
--- a/tests/qemu-iotests/059.out
+++ b/tests/qemu-iotests/059.out
@@ -17,4 +17,11 @@ Formatting 'TEST_DIR/t.IMGFMT', fmt=IMGFMT size=67108864
 L1 size too big
 qemu-io: can't open device TEST_DIR/t.vmdk
 no file open, try 'help open'
+=== Testing monolithicFlat creation and opening ===
+
+Formatting 'TEST_DIR/t.IMGFMT', fmt=IMGFMT size=2147483648
+image: TEST_DIR/t.vmdk
+file format: vmdk
+virtual size: 2.0G (2147483648 bytes)
+disk size: 4.0K
 *** done
-- 
1.7.1

